include/seqan3/core/platform.hpp

```diff
@@ -189,9 +189,27 @@
 //  Deprecation Messages
 // ============================================================================
 
+//!\brief _Pragma requires a string-literal and # makes it a string
+#ifndef SEQAN3_PRAGMA
+#   define SEQAN3_PRAGMA(non_string_literal) _Pragma(#non_string_literal)
+#endif
+
 //!\brief Deprecation message for SeqAn 3.1.0 release.
-#if !defined(SEQAN3_DEPRECATED_310)
-#   define SEQAN3_DEPRECATED_310 [[deprecated("This will be removed in SeqAn-3.1.0; please see the documentation.")]]
+#ifndef SEQAN3_DEPRECATED_310
```

Comment:
why here `ifndef` but in all other places `!defined(...)`?

Reply:
most places are `#ifndef` or `#ifdef`.

Reply:
but not the line below.

Reply:
ah sorry I misunderstood you, this is a leftover from a rewrite, before I had both conditions on one line